Task/build banner modal

rachelfranklin1 opened the pull request:
Hey team 👋 

This PR aims to resolve task #1 

<img width="1427" alt="Screenshot 2019-05-16 08 47 59" src="https://user-images.githubusercontent.com/26777813/57835831-5462ea80-77b7-11e9-9247-d26b9d6251ed.png">

I've added an esc handler and a 'x' button to dismiss the modal. Happy to make any changes; styling or otherwise!

rachelfranklin1 committed:
modal showing onClick of link

rachelfranklin1 committed:
modal showing onClick of link

rachelfranklin1 committed:
toggleModal

rachelfranklin1 committed:
toggleModal

rachelfranklin1 committed:
barebones of VideoModal component

rachelfranklin1 committed:
barebones of VideoModal component

rachelfranklin1 committed:
adding iframed video

rachelfranklin1 committed:
adding iframed video

rachelfranklin1 committed:
remove inline styles / tidy up

rachelfranklin1 committed:
remove inline styles / tidy up

rachelfranklin1 committed:
add close button & styling changes

rachelfranklin1 committed:
add close button & styling changes

rachelfranklin1 committed:
add esc key handler & responsive fix

rachelfranklin1 committed:
add esc key handler & responsive fix

rachelfranklin1 committed:
Update styles.module.scss

rachelfranklin1 committed:
Update styles.module.scss

rachel-harris1 commented on src/components/ChoiceGrid/Styles.module.scss:
```diff
@@ -234,12 +226,12 @@
   text-align: center;
   @include font-size($text-large);
-  @include mq('tablet') {
+  @include mq("tablet") {
```
Apologies for these changes caused by my linter. V annoying!

ThePaulMcBride replied:
My linter does the same thing! We should set up a linting config so we don't have to think about it. I'll try and have a look at this later.

barrymcgee replied:
[Implementing Prettier](https://www.barrymcgee.co.uk/notes/format-your-code-prettier) would solve this issue once and for all. If you guys are interested, I can submit a PR?

shylands replied:
@barrymcgee hey man, yeah all contributions welcome!

rachelfranklin1 committed:
tidy up

rachelfranklin1 committed:
tidy up

rachelfranklin1 commented:
and we have broken checks again 🤦‍♂  i'll try and fix these this afternoon!

ThePaulMcBride commented on src/components/ConvincedButton/index.js:
```diff
@@ -0,0 +1,32 @@
+      <a
+        href="#"
+        title="See how climate change happens"
+        onClick={() => toggleModal()}
```
It might be worth using a button here instead of a link, that way, we don't add a `#` to the URL

ThePaulMcBride commented on src/components/HomeIntro/index.js:
```diff
@@ -61,11 +61,7 @@ const HomeIntro = () => {
-                Need convinced? Watch this
-              </a>
-            </div>
+            <ConvincedButton />
```
I think this needs imported, which is what is causing the error at build time

rachel-harris1 replied:
Ah, I must have accidentally removed it during a rebase. Great spot, thanks!

ThePaulMcBride commented on src/components/VideoModal/index.js:
```diff
@@ -0,0 +1,54 @@
+import Styles from "./styles.module.scss";
+
+export default function VideoModal({ closeClick, closeModal }) {
+  window.onkeydown = function onEscKey(e) {
```
I think we need to wrap this in `useEffect` to make sure we don't keep adding the event listener or accidentally remove it somewhere else. I can help with this if you like, `useEffect` is really damn confusing!

ThePaulMcBride reviewed (changes requested):
This looks damn good!

Have a quick look at the z-index on the modal as I think it sometimes gets hidden behind the section below it.

![Screenshot 2019-05-16 at 09 23 41](https://user-images.githubusercontent.com/6891650/57838879-77909880-77bd-11e9-8ed7-76590534fb2a.png)

I can't wait to get this launched!

rachelfranklin1 committed:
bad merge

rachelfranklin1 committed:
bad merge

rachelfranklin1 committed:
Update index.js

rachelfranklin1 committed:
Update index.js

rachelfranklin1 committed:
small styling change

rachelfranklin1 committed:
small styling change

shylands commented:
Looks good! The Z index issue is because there's a z index on choiceGridContainer. Should be easy to sort by changing the z index values.

A couple other small things:
1) Could we put a faded background colour behind the modal when it's open?
2) In the iOS simulator, the X close button isn't appearing
3) Not a big deal, but for some reason, the modal contents are slightly blurry on my external modal. When I turn the padding off it's fine, some kind of scaling issue? See img below:

![ClimateChoice 2019-05-16 17-55-44](https://user-images.githubusercontent.com/310547/57868795-f3153880-7803-11e9-8da6-2dc9ca740210.png)

rachelfranklin1 committed:
change anchor element to button

rachelfranklin1 committed:
start of useEffect

rachelfranklin1 committed:
update z-index

rachelfranklin1 committed:
better z-index

rachelfranklin1 committed:
fix scaling issue & rework open/close modal functions

rachelfranklin1 committed:
fix close button on mobile

rachelfranklin1 committed:
useEffect clean up

rachelfranklin1 commented:
@ThePaulMcBride I've added the esc handler in `useEffect` 🙌🏼thanks for the review!

@shylands I've fixed your points (although I couldn't replicate the blurry issue, but pretty confident I know why it was being caused), thank you! 

The only thing I couldn't figure out was how to fade the page background when the modal is open. As React nest's components, the modal was also fading as it was taking the styles of it's parent 😞 

If you have any idea how I could go about this resolving this, please let me know. Happy to implement!

rachelfranklin1 committed:
fix console warnings

shylands committed:
style improvements

shylands commented:
@rachelfranklin1 Looking good! I've merged it as I think the only thing missing still is a faded BG and it's not that important. Cheers!
